src/misc/messages.cpp

```diff
@@ -251,33 +251,98 @@ bool MSG_Write(const char * location) {
 
 // 2. It also supports the more convenient syntax without needing to provide a
 //    filename or path: `-lang ru`. In this case, it constructs a path into the
-//    platform's config path/translations/<lang>[.utf8].lng.
+//    platform's config path/translations/<lang>.lng.
 
 void MSG_Init([[maybe_unused]] Section_prop *section)
 {
-	// TODO: After migration to C++20 try to switch to constexpr
-	static const std_fs::path subdir   = "translations";
-	static const std::string extension = ".lng";
+	static const std_fs::path subdirectory = "translations";
+	static const std::string extension     = ".lng";
+
+	const auto& host_language = GetHostLanguage();
 
-	const auto lang = control->GetLanguage();
+	// Get the language file from command line
+	assert(control);
+	auto language_file = control->GetArgumentLanguage();
 
-	// If the language is english, then use the internal message
-	if (lang.empty() || lang.starts_with("en")) {
+	// If not available, get it from the config file
+	if (language_file.empty()) {
+		const auto section = control->GetSection("dosbox");
+		assert(section);
+		language_file = static_cast<const Section_prop*>(section)->Get_string(
+		        "language");
+	}
+
+	// If the language is English, use the internal messages
+	if (language_file == "en" || language_file == "en.lng") {
 		LOG_MSG("LANG: Using internal English language messages");
 		return;
 	}
 
-	bool result = false;
-	if (lang.ends_with(".lng"))
-		result = load_message_file(get_resource_path(subdir, lang));
-	else
-		// If a short-hand name was provided then add the file extension
-		result = load_message_file(get_resource_path(subdir, lang + extension));
+	std_fs::path file_path = {};
+
+	// If concrete language file is provided, try to load it
+	if (!language_file.empty() && language_file != "auto") {
+		if (language_file.ends_with(extension)) {
+			file_path = get_resource_path(subdirectory, language_file);
+		} else {
+			// If a short-hand name was provided,
+                        // add the file extension
+			file_path = get_resource_path(subdirectory,
+			                            language_file + extension);
+		}
+
+		const auto result = load_message_file(file_path);
+		if (!result) {
+			LOG_MSG("LANG: Could not load language file '%s', "
+			        "using internal English language messages",
+			        language_file.c_str());
+		} else {
+			LOG_MSG("LANG: Loaded language file '%s'",
+			        file_path.string().c_str());
```

Comment:
I noticed you use both the `LANG` and `LOCALE` log prefixes. Do we need both? Would it make sense to settle for either only?

Reply:
Changed everything to `LOCALE`.